wicket-orientdb/src/main/java/ru/ydn/wicket/wicketorientdb/utils/query/StringQueryManager.java

```diff
@@ -91,11 +92,13 @@ private String getFilterCriteria(String sql) {
 		if (!managers.isEmpty()) {
 			String filter = applyFilters();
 			if (!Strings.isNullOrEmpty(filter)) {
-				boolean containsWhere = sql.toUpperCase().contains("WHERE");
-				if (containsWhere) sb.append(" AND(");
-				else sb.append(" WHERE ");
-				sb.append(filter);
-				if (containsWhere) sb.append(")");
+				Matcher matcher = EMBEDDED_PATTERN.matcher(sql);
```

Comment:
Please see below. Not sure that this wrapping to one more select should be done on input level. Imho it should be done internally during prepareSql().